### ISSUE TITLE
Internal server error from Ipsilon when calling user_getfield()

### ISSUE THREAD

**Issue by ncoghlan** [puiterwijk/flask-oidc#24]
Working on Anitya's OIDC integration, I'm sending the access token to the server as part of the request's query string. The token validation decorator correctly finds it there, and sets the flask global `g.oidc_token_info`. However, the global that the rest of flask-oidc is looking for is `g.oidc_id_token`, and it never checks `g.oidc_token_info` at all (even as a fallback), so I need to pass the access token explictly to the API as follows: `APP.oidc.user_getfield('user_id', access_token)`.

`access_token` is retrieved from the request using flask-reqparse:

    _BASE_ARG_PARSER = reqparse.RequestParser(trim=True, bundle_errors=True)
    _BASE_ARG_PARSER.add_argument('access_token', type=str)

And then in the request itself, the initial parser is created via `_BASE_ARG_PARSER.copy()`

But when I do that, the call to `user_getfield` fails with the following traceback:

```
  File "/home/ncoghlan/fedoradevel/anitya/anitya/api_v2.py", line 65, in post
    user_id=APP.oidc.user_getfield('user_id', access_token),
  File "/home/ncoghlan/.virtualenvs/anitya/lib/python3.5/site-packages/flask_oidc/__init__.py", line 167, in user_getfield
    info = self.user_getinfo([field], access_token)
  File "/home/ncoghlan/.virtualenvs/anitya/lib/python3.5/site-packages/flask_oidc/__init__.py", line 196, in user_getinfo
    all_info = self._retrieve_userinfo(access_token)
  File "/home/ncoghlan/.virtualenvs/anitya/lib/python3.5/site-packages/flask_oidc/__init__.py", line 241, in _retrieve_userinfo
    info = _json_loads(content)
  File "/home/ncoghlan/.virtualenvs/anitya/lib/python3.5/site-packages/flask_oidc/__init__.py", line 52, in _json_loads
    return json.loads(content)
  File "/usr/lib64/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.5/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

The debug logs show the following response from Ipsilon:

```
flask_oidc: DEBUG: Retrieved user info: b'<!DOCTYPE html>\n<!--[if IE 8]><html class="ie8 login-pf"><![endif]-->\n<!--[if gt IE 8]><!-->\n<html class="login-pf">\n<!--<![endif]-->\n  <head>\n    <title>Internal Error</title>\n    <meta charset="UTF-8">\n    <meta name="viewport" content="width=device-width, initial-scale=1.0">\n    <link href="/ui/css/ipsilon.css" rel="stylesheet" media="screen, print">\n    <link href="/ui/css/styles.css" rel="stylesheet" media="screen, print">\n  </head>\n  <body>\n    <a href="/" id="badge" tabindex="-1">\n      <img src="/ui/img/logo.svg" alt="Ipsilon IdP" />\n    </a>\n    <div class="container">\n      <div class="row">\n        <div class="col-sm-12">\n          <div id="brand">\n            <img src="/ui/img/brand-lg.png" alt="Ipsilon">\n          </div>\n        </div>\n        \n<div class="col-sm-12">\n  <h1>500 - Internal Server Error</h1>\n  \n    <p>Ipsilon encountered an unexpected internal error while trying to\n       fulfill your request.</p>\n  \n  <p>Please retry again.</p>\n  <p>If the error persists, contact the server administrator to resolve\n     the problem.</p>\n</div>\n\n      </div>\n    </div>\n  </body>\n</html>'
```

**Comment by puiterwijk** [comment]
The 500 was a bug on the Ipsilon side caused by flask-oidc not providing the access token, which was fixed in 4e31d6a0300a97896bde0b8e9805f84558d96177.

**Comment by ncoghlan** [comment]
@puiterwijk tracked this down to a missing Content-type header when submitting the userinfo request to Ipsilon (so the server wasn't decoding the access_token correctly)
